### ISSUE TITLE
Terraform Crashing on validation (terraform plan/apply)

### ISSUE THREAD

**Issue by saianaparthi** [hashicorp/terraform#26048]
### Terraform Version

```
Terraform v0.13.1
+ provider registry.terraform.io/hashicorp/azurerm v2.20.0
+ provider registry.terraform.io/hashicorp/helm v1.2.4
```

### Terraform Configuration Files

```main.tf
# ----------------------------------------------------------------------------------------------------------------------
# REQUIRE A SPECIFIC TERRAFORM VERSION OR HIGHER
# This module has been updated with 0.12 syntax, which means it is no longer compatible with any versions below 0.12.
# ----------------------------------------------------------------------------------------------------------------------

terraform {
  required_version = ">= 0.13"
  backend "azurerm" {
    resource_group_name   = "some_rg"
    storage_account_name  = "some_sa"
    container_name        = "some_container_name"
    subscription_id       = "some_subscription_id"
    tenant_id             = "some_tenant_id"
    client_id             = "some_client_id"
    client_secret         = "some_client_secret"
    key                   = "some_key"
  }
}

provider "azurerm" {
  version = "~> 2.20.0" #"~> 1.39"
  features {}
  client_id       = local.workspace["deployment_client_id"]
  subscription_id = local.workspace["deployment_subscription_id"]
  tenant_id       = local.workspace["deployment_tenant_id"]
  client_secret   = local.workspace["deployment_client_secret"]
}
```

### Crash Output

module.kyron-helm.data.azurerm_kubernetes_cluster.k8s: Refreshing state...
panic: not a collection type

goroutine 824 [running]:
        /home/circleci/project/project/vendor/github.com/zclconf/go-cty/cty/collection.go:25 +0xc1
github.com/hashicorp/terraform/terraform.createEmptyBlocks(0xc000e41f80, 0x2d39220, 0xc000aec0d8, 0x2355700, 0xc000eec3f0, 0x2d39220, 0xc000aec260, 0x2355700, 0xc000f52ba0)
github.com/hashicorp/terraform/terraform.dataObjectHasChanges(0xc000e41f80, 0x2d39220, 0xc000aec0d8, 0x2355700, 0xc000eec3f0, 0x2d39220, 0xc000aec260, 0x2355700, 0xc000f52ba0, 0x0, ...)
        /home/circleci/project/project/terraform/eval_read_data_plan.go:211 +0x1a5
github.com/hashicorp/terraform/terraform.(*evalReadDataPlan).Eval(0xc0001e12c0, 0x2d7b6e0, 0xc001a01ad0, 0x1, 0x1, 0x0, 0x0)
        /home/circleci/project/project/terraform/eval_read_data_plan.go:107 +0x123f
github.com/hashicorp/terraform/terraform.EvalRaw(0x2cd1460, 0xc0001e12c0, 0x2d7b6e0, 0xc001a01ad0, 0x0, 0x0, 0x0, 0x0)
        /home/circleci/project/project/terraform/eval.go:49 +0xce
github.com/hashicorp/terraform/terraform.(*EvalSequence).Eval(0xc000d1a4c0, 0x2d7b6e0, 0xc001a01ad0, 0x1, 0x1, 0x6, 0xc0001b9670)
        /home/circleci/project/project/terraform/eval_sequence.go:20 +0x104
github.com/hashicorp/terraform/terraform.EvalRaw(0x2cd0ec0, 0xc000d1a4c0, 0x2d7b6e0, 0xc001a01ad0, 0x230aac0, 0x40098a2, 0x226ae60, 0xc000a42cd0)
        /home/circleci/project/project/terraform/eval.go:49 +0xce
github.com/hashicorp/terraform/terraform.Eval(0x2cd0ec0, 0xc000d1a4c0, 0x2d7b6e0, 0xc001a01ad0, 0xc000d1a4c0, 0x2cd0ec0, 0xc000d1a4c0, 0x0)
        /home/circleci/project/project/terraform/eval.go:35 +0x54
github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x2699c60, 0xc000a42290, 0x0, 0x0, 0x0)
        /home/circleci/project/project/terraform/graph.go:73 +0xc7e
github.com/hashicorp/terraform/dag.(*Walker).walkVertex(0xc0001b9560, 0x2699c60, 0xc000a42290, 0xc000a62740)
        /home/circleci/project/project/dag/walk.go:387 +0x35e
created by github.com/hashicorp/terraform/dag.(*Walker).Update
        /home/circleci/project/project/dag/walk.go:309 +0x11de

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

SECURITY WARNING: the "crash.log" file that was created may contain
sensitive information that must be redacted before it is safe to share
on the issue tracker.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
### Steps to Reproduce

terraform init
terraform apply

**Comment by nexxai** [comment]
This bug has already been fixed [here](https://github.com/hashicorp/terraform/pull/26028) and will be included in v13.2

**Comment by jbardin** [comment]
Closing as duplicate of #26011
